src/dataset-builder/DomainCriteriaSelector.test.ts

```diff
@@ -47,10 +47,11 @@ describe('DomainCriteriaSelector', () => {
     await user.click(screen.getByLabelText('add'));
     await user.click(screen.getByText('Add to group'));
     // Assert
+    const expectedCriteria = toCriteria(domainOption)(concept);
+    // Update expected index; need to subtract one so indexes match.
+    expectedCriteria.index -= 1;
```

Comment:
I see your comment, but why is this necessary?

Reply:
Every time a new criteria is created it's assigned a unique index. So each call to `toCriteria()` will result in a new unique index (which is the previous index plus one). The code subtracts one to "back up" the counter so the expected data matches the actual data.

A better approach would be to mock the index creation method to control its results; I can look at doing that for this test.

Reply:
I was able to get rid of the `- 1` by stubbing the criteria index creation function in the test.